src/contexts/globalLayout/GlobalLayoutContext.tsx

```diff
@@ -0,0 +1,371 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
```

Comment:
this context is brilliant 🙌🙌 
great job figuring this out! love the idea of handling everything in one place